packages/components/src/components/Modal/Modal.docs.mdx

```diff
@@ -0,0 +1,201 @@
+
+Okno v popredí. Obsah stráky sa zobrazuje akoby za ním.
+
+Poskytuje dôležité informácie, vyžaduje potvrdenie akcie, vybratie možnosti zo zoznamu, a podobne. Keď je dialógové okno prítomné, zabraňuje použitiu aplikácie až kým ho používateľ potvrdí, zruší alebo vykoná požadovanú akciu.
+
+**Upozornenie:** Keďže dialógové okná úmyselne prerušujú prácu, mali by byť používané len vo výnimočných prípadoch.
+
+### Ukážka
+
+<Preview>
+  <Button data-a11y-dialog-show="example-modal">Zobraziť ukážku</Button>
+  <Modal id="example-modal" header="Lorem Ipsum">
+    <p className="no-mrg-bottom">{modalPlaceholderText}</p>
+  </Modal>
+</Preview>
+
+### Použitie
+
+Zobrazenie dialógového okna môže byť ovládané dvoma spôsobmi – explicitne cez `isVisible` prop alebo pomocou metód `show` a `hide` dostupných cez referenciu. Ovladanie cez referenciu nie je odporúčané a može byť odstránené v budúcnosti.
+
+```typescript
+const ControlledModal = () => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  return (
+    <Modal
+      header="Dobrý deň"
+      isVisible={isVisible}
+      onHide={() => setIsVisible(false)}
+    >
+      ...
+    </Modal>
+  );
+};
+```
+
+```typescript
+const UncontrolledModal = () => {
+  const modalRef = useRef<Modal["ref"]>(null);
+
+  const show = () => modalRef.current?.show();
+  const hide = () => modalRef.current?.hide();
+
+  return (
+    <Modal ref={modalRef} header="Dobrý deň">
+      ...
+    </Modal>
+  );
+};
+```
+
+### Lazy inicializácia modalu
+
+V prípade, že je obsah modalu možné inicializaovať až v momente zobrazovania, je možné kombinovať kontrolovaný modal s podmienkov na zobrazenie.
+
+```typescript
+const ControlledLazyModal = () => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  return (
+    isVisible && (
+      <Modal
+        header="Dobrý deň"
+        isVisible={isVisible}
+        onHide={() => setIsVisible(false)}
+      >
+        ...
+      </Modal>
+    )
+  );
+};
+```
+
+### Varianty
+
+#### Základná
+
+<Preview>
+  <Button data-a11y-dialog-show="default-modal">Zobraziť modál</Button>
+  <Modal id="default-modal" header="Základný modál">
+    <p className="no-mrg-bottom">{modalPlaceholderText}</p>
+  </Modal>
+</Preview>
+
+_Poznámka:_ Ak je na stránke viacero aktívnych modálov, posledný aktivovaný modál bude navrchu.
+
+#### Dlhý obsah
+
+Ak modál obsahuje dlhý text, napr. modál pre súhlas s podmienkami používania, zvážte font Amalia pre obsah použitím triedy `.text-font-secondary`.

```

Comment:
Tu je to specifike pre DDS - specificke veci dajme prosim prec - ako napr. font. Pozri prosim celu dokumentaciu, ci tam niesu dalsie taketo veci.